File: src/main/java/com/github/kuro46/embedscript/EmbedScriptPlugin.java
package com.github.kuro46.embedscript;

import org.bukkit.Bukkit;
import org.bukkit.configuration.InvalidConfigurationException;
import org.bukkit.plugin.java.JavaPlugin;

import java.io.IOException;
import java.util.logging.Level;

/**
 * @author shirokuro
 */
public class EmbedScriptPlugin extends JavaPlugin {
    @Override
    public void onEnable() {
        long begin = System.currentTimeMillis();
        try {
            EmbedScript.initialize(this);
        } catch (IOException | InvalidConfigurationException | IllegalStateException e) {
            getLogger().log(Level.SEVERE, "Failed to initialize the plugin! disabling...", e);
            Bukkit.getPluginManager().disablePlugin(this);
            return;
        }
        long end = System.currentTimeMillis();
        getLogger().info(String.format("Enabled! (%sms)", end - begin));
    }

    @Override
    public void onDisable() {
        EmbedScript.reset();
    }
}

File: src/main/kotlin/com/github/kuro46/embedscript/EmbedScriptPlugin.kt
package com.github.kuro46.embedscript

import java.util.logging.Level
import org.bukkit.Bukkit
import org.bukkit.plugin.java.JavaPlugin

/**
 * @author shirokuro
 */
class EmbedScriptPlugin : JavaPlugin() {
    override fun onEnable() {
        val begin = System.currentTimeMillis()

        val initResult = runCatching { EmbedScript.initialize(this) }
        initResult.exceptionOrNull()?.let {
            logger.log(Level.SEVERE, "Failed to initialize the plugin! disabling...", it)
            Bukkit.getPluginManager().disablePlugin(this)
        } ?: run {
            val end = System.currentTimeMillis()
            logger.info("Enabled! (${end - begin}ms)")
        }
    }

    override fun onDisable() {
        EmbedScript.reset()
    }
}
